### PR TITLE
PR: IntuneDeviceConfigurationIdentityProtectionPolicyWindows10 - Initial release

### PR DESCRIPTION
**Pull request by William-Francillette**
#### This Pull Request (PR) fixes the following issues

* IntuneDeviceConfigurationIdentityProtectionPolicyWindows10
  * Initial release
    FIXES [#2831](https://github.com/microsoft/Microsoft365DSC/issues/2831)

### CODE REVIEW

**Comment by ykuijs** [comment]
In this PR, the example is a psm1 file instead of a ps1 file. Please change this into a ps1 file.

**Comment by William-Francillette** [comment]
@ykuijs same fix as IntuneEndpointProtection

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 [hunk 1]
```diff
@@ -0,0 +1,1538 @@
...
+        New-M365DSCLogEntry -Message 'Error during Export:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return ''
+    }
+}
+    function Update-DeviceConfigurationPolicyAssignment
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:777]: The indentation of this function is off. You use the formatting functionality of Visual Studio Code (Alt-Shift-F) to correct the formatting automatically.

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 [hunk 2]
```diff
@@ -0,0 +1,1538 @@
...
+
+            [Parameter()]
+            [ValidateSet('v1.0','beta')]
+            [System.String]
+            $APIVersion='beta'
+        )
+        try
+        {
+            $deviceManagementPolicyAssignments=@()
+
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:802]: Please remove the additional empty line

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 [hunk 3]
```diff
@@ -0,0 +1,1538 @@
...
+                if($target.deviceAndAppManagementAssignmentFilterId)
+                {
+                    $formattedTarget.Add('deviceAndAppManagementAssignmentFilterId',$target.deviceAndAppManagementAssignmentFilterId)
+                }
+                $deviceManagementPolicyAssignments+=@{'target'= $formattedTarget}
+            }
+            $body=@{'assignments'=$deviceManagementPolicyAssignments}|ConvertTo-Json -Depth 20
+            #write-verbose -Message $body
+            Invoke-MgGraphRequest -Method POST -Uri $Uri -Body $body -ErrorAction Stop
+
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:829]: Please remove the additional empty line

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 [hunk 4]
```diff
@@ -0,0 +1,1538 @@
...
+                -Exception $_
+                -Source $($MyInvocation.MyCommand.Source)
+                -TenantId $TenantId
+                -Credential $Credential
+
+            return $null
+        }
+
+
+    }function Rename-M365DSCCimInstanceParameter
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:843]: Please place the function on a new line and place an empty line between the closing brace and the function

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 [hunk 5]
```diff
@@ -0,0 +1,1538 @@
...
+                $hashProperties.add($keyName,(Rename-M365DSCCimInstanceParameter $property))
+            }
+        }
+        $result = $hashProperties
+    }
+
+    return $result
+    #endregion
+}
+function Get-M365DSCDRGComplexTypeToHashtable
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:904]: Please add an empty line between the closing brace and the function

**Review by ykuijs — CHANGES_REQUESTED** [review]
Some comments

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 (hunk 1 above)
Comment (William-Francillette) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1]: done

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 (hunk 3 above)
Comment (William-Francillette) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:793]: done

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 (hunk 5 above)
Comment (William-Francillette) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:903]: done

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 (hunk 4 above)
Comment (William-Francillette) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1]: done

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1 (hunk 2 above)
Comment (William-Francillette) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1:785]: done

**Comment by William-Francillette** [comment]
> Some comments

Should be good to go now

**Review by ykuijs — APPROVED** [review]
LGTM

**Pull request merged**